nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/processors/script/ExecuteScript.java

```diff
@@ -303,4 +423,57 @@ public void stop() {
 
         return results;
     }
+
+    private void initializeScriptingComponentHelper() {
+        synchronized (scriptingComponentHelper.isInitialized) {
+            if (!scriptingComponentHelper.isInitialized.get()) {
+                scriptingComponentHelper.createResources();
+                scriptingComponentHelper.addDescriptor(ScriptingComponentUtils.USE_DYNAMIC_RELATIONSHIPS);
+            }
+        }
+    }
+
+    private static boolean propertySpecifiesRelationship(final String propertyDescriptorName) {
+        return propertyDescriptorName != null
+                && propertyDescriptorName.startsWith(DYNAMIC_RELATIONSHIP_PREFIX);
+    }
+
+    private static String getRelationshipName(final String propertyDescriptorName) {
+        return !propertySpecifiesRelationship(propertyDescriptorName) || DYNAMIC_RELATIONSHIP_PREFIX.equals(propertyDescriptorName)
+               ? ""
+               : propertyDescriptorName.substring(DYNAMIC_RELATIONSHIP_PREFIX.length());
+    }
+
+    private static boolean isValidRelationshipName(final String relationshipName) {
+        final String nameInLowerCase = relationshipName.toLowerCase();
+        return !(REL_SUCCESS.getName().equals(nameInLowerCase)
+                || REL_FAILURE.getName().equals(nameInLowerCase));
+    }
+
+    private static class RelationshipInvalidator implements Validator {
```

Comment:
This is a nitpick, but I'm not sure we have any other "Invalidator" implementations, usually they're called "Validator" even if their only job is to check conditions that may make it invalid.